enhancements/kubelet/split-filesystem.md

```diff
@@ -0,0 +1,282 @@
+
+This enhancement focuses on enablement in Openshift.
+
+## Motivation
+
+See [KEP Motivation](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/4191-split-image-filesystem/README.md#motivation)
+
+### Definitions
+
+### User Stories
+
+As an Openshift admin, I want to store images in a separate filesystem from ephemeral storage and the writeable layer.
+The images can be on a read-only filesystem while ephemeral storage and the writeable layers can live on a writeable filesystem.
+
+### Goals
+
+- Enable ability to split filesystem in openshift
+- Automate the setup of this feature to avoid user errors
+
+### Non-Goals
+
+- We will not support a separate disk for the entire container runtime filesystem due
+to a lack of interest from customer requests.
+
+- Customers will not have the ability to change the location of the image cache.
+
+### Sketch of what happens when this feature is enabled
+
+- A user specifies in the container runtime config that they want to split the filesystem (use this feature)
+  - Feature gate is set for kubelet
+  - container storage uses image store (hard coded to /var/lib/images)
+  - /var/lib/images is relabeled to match the same selinux labels as /var/lib/container/storage.
+
+### Manual Enablement of Feature
+
+In the developer preview, a user can run the following steps to enable this feature.
+
+We will automate these steps for tech preview.
+
+#### Feature gate
+
+User needs to set `KubeletSeparateDiskGC` feature gate in the kubelet config.
+
+#### Storage Configuration
+
+One could use a butane config as follows:
+
+```storage.bu
+variant: openshift
+version: 4.14.0
+metadata:
+  name: 40-storage-override
+  labels:
+    machineconfiguration.openshift.io/role: worker
+storage:
+  files:
+  - path: /etc/containers/storage.conf
+    mode: 0644
+    overwrite: true
+    contents:
+      inline: |
+        [storage]
+        
+        # Default Storage Driver
+        driver = "overlay"
+        
+        runroot = "/var/containers/storage"
+        graphroot = "/var/lib/containers/storage"
+        imageroot = "/var/lib/images"
+```
+
+And then run `butane storage.bu -o storage.yaml
+
+Applying storage.yaml will apply this machine config to your workers.
+
+#### Labeling Filesystem
+
+One could use the following systemd file to relabel the imagestore location.
+
+```
+        [Unit]
+        Description=Label ImageStore
+        After=crio-install.service
+
+        [Service]
+        Type=oneshot
+        ExecStart=rpm-ostree install \
+          -y \
+          --apply-live \
+          --allow-inactive \
+          policycoreutils-python-utils
+        ExecStart=semanage fcontext -a -e /var/lib/containers/storage /var/lib/images
+        ExecStart=restorecon -R -v /var/lib/images
+
+        [Install]
+        WantedBy=multi-user.target
+```
+
+#### Remove all old images
+
+Since the image cache has changed locations, all the old images left over should be removed.
```

Comment:
Why would disconnected installers be impacted here?

Reply:
we wouldn't be able to GC old images if crio doesn't keep track of the images at the old location

Reply:
I asked podman about what they do is this scenario. Generally once these configurations change, the containers/images are not cleaned up with any podman functions.

I think day 1 support for this would be necessary. Maybe we could state that if one enables this feature, deleting the graphroot/imagestore is the recommended procedure. I don't know if that means we need to do this with MCO.

Reply:
the more I think about this problem the more i think we should hardcode imagestore to `/var/lib/images` permanently and handle the update once for everyone then not have to deal with it again